lib/Minz/Log.php

```diff
@@ -70,12 +73,31 @@ public static function record ($information, $level, $file_name = null) {
 			     . ' [' . $level_label . ']'
 			     . ' --- ' . $information . "\n";
 
+			self::checkLogfileSize($file_name);
+
 			if (file_put_contents($file_name, $log, FILE_APPEND | LOCK_EX) === false) {
 				throw new Minz_PermissionDeniedException($file_name, Minz_Exception::ERROR);
 			}
 		}
 	}
 
+	/**
+	 * Make sure we do not waste a huge amount of disk space with old log messages.
+	 *
+	 * This method can be called multiple times for one script execution, but its result will not change unless
+	 * you call clearstatcache() in between. We won't due do that for performance reasons.
+	 *
+	 * @param $file_name
+	 * @throws Minz_PermissionDeniedException
+	 */
+	protected static function checkLogfileSize($file_name) {
+		if (file_exists($file_name) && filesize($file_name) > self::MAX_LOG_SIZE) {
+			if (!unlink($file_name)) {
```

Comment:
We need to truncate the file, not delete it :-)

Reply:
I would like to keep the newest messages.
I think the algorithm could be to divide the logfile by 2 when it increases above a threshold (delete first half of the file).
It is probably not needed to care about keeping entire lines.